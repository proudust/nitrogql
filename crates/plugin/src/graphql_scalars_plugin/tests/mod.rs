#![cfg(test)]

use std::collections::HashMap;

use crate::{GraphQLScalarsPlugin, PluginV1Beta};

use insta::assert_snapshot;

#[test]
fn schema_emission() {
    let mut p = GraphQLScalarsPlugin::default();
    let extensions: HashMap<String, HashMap<String, serde_yaml::Value>> = serde_yaml::from_str(
        r#"{
    "Date": {
        "nitrogql:kind": "scalar",
        "codegenScalarType": "Date"
    },
    "DateTime": {
        "nitrogql:kind": "scalar",
        "codegenScalarType": "number"
    },
    "Other": {
        "nitrogql:kind": "object",
        "codegenScalarType": "string"
    }
}"#,
    )
    .unwrap();
    p.load_schema_extensions(crate::PluginSchemaExtensions {
        type_extensions: &extensions,
    });
    assert_snapshot!(p.schema_addition().unwrap());
}
