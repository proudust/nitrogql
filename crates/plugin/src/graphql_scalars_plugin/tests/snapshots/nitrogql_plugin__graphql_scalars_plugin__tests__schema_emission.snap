---
source: crates/plugin/src/graphql_scalars_plugin/tests/mod.rs
expression: p.schema_addition().unwrap()
---
extend scalar Date @nitrogql_ts_type(type: "Date")
extend scalar DateTime @nitrogql_ts_type(type: "number")

